app/lib/features/onboarding/pages/analytics_opt_in_page.dart

```diff
@@ -74,13 +78,58 @@ class AnalyticsOptInPage extends ConsumerWidget {
   }
 
   Widget _buildSkipActionButton(BuildContext context) {
-    return ActerPrimaryActionButton(
+    return ActerPrimaryActionButton(                
       key: AnalyticsOptInPage.skipBtn,
-      onPressed: () => context.goNamed(Routes.main.name),
+      onPressed: () => _onSkipPressed(context),
       child: Text(
         L10n.of(context).done,
         style: Theme.of(context).textTheme.bodyMedium,
       ),
     );
   }
+
+  Future<void> _onSkipPressed(BuildContext context) async {
+    if (Platform.isAndroid) {
```

Comment:
as pointed out above, this shouldn't happen in this specific page but instead be moved to a single instance that deals with all the missing states upon login or startup and compiles the list of pages that need to be dealt with.

As the overall requests shows why this is a better approach. This needs to move ...